var/spack/repos/builtin/packages/fenics-dolfin/package.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class FenicsDolfin(CMakePackage):
+    """DOLFIN is the C++/Python interface of FEniCS."""
+
+    homepage = "http://fenicsproject.org/"
+    git      = "https://bitbucket.org/fenics-project/dolfin.git"
+
+    def url_for_version(self, version):
```

Comment:
> I prefer to keep all functions together. So I would move `url_for_version` above `cmake_is_on`.

Should I move `url_for_version` after `depends_on` in the other packages as well to be consistent or only in fenics-dolfin?

> Also, you should keep a `url` for the newest version so that `spack versions` and `spack checksum` work.

Sorry, I did notice that in a discussion for another package somewhere but I completely forgot about it.

Reply:
> Should I move `url_for_version` after `depends_on` in the other packages as well to be consistent or only in fenics-dolfin?

I think this would look nicer because `url_for_version` is just a workaround for old naming conventions and could be removed at some stage so putting it at the end of those other packages might be best. What do you think @adamjstewart ?

Reply:
My statement applies to all packages, not just `fenics-dolfin`.